torch/distributed/distributed_c10d.py

```diff
@@ -1393,6 +1393,15 @@ def all_gather_object(object_list, obj, group=group.WORLD):
         known to be insecure. It is possible to construct malicious pickle data
         which will execute arbitrary code during unpickling. Only call this
         function with data you trust.
+
+    Example::
+        >>> # Note: Process group initialization omitted on each rank.
+        >>> import torch.distributed as dist
+        >>> gather_objects = ["foo", 12, {1: 2}] # any picklable object
```

Comment:
this assumes `world_size==3`?

Reply:
Right, I guess we could just add this as a comment to keep the example clear and concise.